dataset/src/main/kotlin/com/cosmotech/dataset/service/DatasetServiceImpl.kt

```diff
@@ -127,9 +129,15 @@ class DatasetServiceImpl(
     private val unifiedJedis: UnifiedJedis,
     private val csmRbac: CsmRbac,
     private val csmAdmin: CsmAdmin,
-    private val resourceScanner: ResourceScanner
+    private val resourceScanner: ResourceScanner,
+    @Value("\${csm.platform.twincache.useGraphModule}") private val useGraphModule: Boolean
 ) : CsmPhoenixService(), DatasetApiServiceInterface {
 
+  private val notImplementedExceptionMessage =
+      "The API is not configured to use Graph functionnalities. " +
+          "This endpoint is deactivated so. " +
```

Comment:
Not sure what the ` so` at the end of the sentence is for. At least it feels weird English to me.